applications/MeshingApplication/custom_utilities/uniform_refine_utility.cpp

```diff
@@ -308,7 +309,7 @@ void UniformRefineUtility<TDim>::CreateNodeInEdge(
         // Add the node to the sub model parts
         const int key0 = mNodesColorMap[rEdge(0)->Id()];
         const int key1 = mNodesColorMap[rEdge(1)->Id()];
-        const int key = mSubModelPartsColors.IntersectKeys(key0, key1, mColors);
+        const int key = mIntersections[std::minmax(key0, key1)];
```

Comment:
Are you using negative values? (otherwise it could be unsigned int or std::size_t)

Reply:
I am using positive values, a color. I add it to my todo list